lib/Service/ImageService.php

```diff
@@ -0,0 +1,676 @@
+		$this->previewManager = $previewManager;
+	}
+
+	/**
+	 * Get image content or preview from file id
+	 * @param int $documentId
+	 * @param string $imageFileName
+	 * @param string $userId
+	 * @return File|\OCP\Files\Node|ISimpleFile|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getImage(int $documentId, string $imageFileName, string $userId) {
+		$textFile = $this->getTextFile($documentId, $userId);
+		return $textFile === null ? null : $this->getImagePreview($imageFileName, $textFile);
+	}
+
+	/**
+	 * Get image content or preview from file id in public context
+	 * @param int $documentId
+	 * @param string $imageFileName
+	 * @param string $shareToken
+	 * @return File|\OCP\Files\Node|ISimpleFile|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getImagePublic(int $documentId, string $imageFileName, string $shareToken) {
+		$textFile = $this->getTextFilePublic($documentId, $shareToken);
+		return $textFile === null ? null : $this->getImagePreview($imageFileName, $textFile);
+	}
+
+	/**
+	 * @param string $imageFileName
+	 * @param File $textFile
+	 * @return File|\OCP\Files\Node|ISimpleFile|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	private function getImagePreview(string $imageFileName, File $textFile) {
+		$attachmentFolder = $this->getOrCreateAttachmentDirectoryForFile($textFile);
+		if ($attachmentFolder !== null) {
+			try {
+				$imageFile = $attachmentFolder->get($imageFileName);
+			} catch (NotFoundException $e) {
+				return null;
+			}
+			if ($imageFile instanceof File) {
+				if ($this->previewManager->isMimeSupported($imageFile->getMimeType())) {
+					return $this->previewManager->getPreview($imageFile, 1024, 1024);
+				} else {
+					return $imageFile;
+				}
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Save an uploaded image in the attachment folder
+	 *
+	 * @param int $documentId
+	 * @param string $newFileName
+	 * @param string $newFileContent
+	 * @param string $userId
+	 * @return array
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function uploadImage(int $documentId, string $newFileName, string $newFileContent, string $userId): array {
+		$textFile = $this->getTextFile($documentId, $userId);
+		if (!$textFile->isUpdateable()) {
+			throw new Exception('No write permissions');
+		}
+		$saveDir = $this->getOrCreateAttachmentDirectoryForFile($textFile);
+		if ($saveDir !== null) {
+			$fileName = (string) time() . '-' . $newFileName;
+			$savedFile = $saveDir->newFile($fileName, $newFileContent);
+			return [
+				'name' => $fileName,
+				'id' => $savedFile->getId(),
+				'documentId' => $textFile->getId(),
+			];
+		} else {
```

Comment:
From PhpStorm: [EA] Child instructions can be extracted here (clearer intention, lower complexity numbers).